documentation/puppet-api/v3/environment_modules.markdown

```diff
@@ -107,11 +152,69 @@ HTTP/1.1 400 Bad Request
 The environment must be purely alphanumeric, not 'bog|us'
 ```
 
+### No metadata.json file
+
+If your modules do not have a [metadata.json](https://docs.puppet.com/puppet/4.9/modules_metadata.html)
```

Comment:
The `/latest/` URL should be fine as long as Puppet and Server are released together.